yaqc_cmds/_plot.py

```diff
@@ -94,19 +95,28 @@ def on_channels_changed(self):
         self.channel.set_allowed_values(new)
 
     def on_data_file_created(self):
-        data = somatic._wt5.get_data_readonly()
-        self.axis.set_allowed_values(data.axis_expressions)
+        if self.data is not None:
+            self.data.close()
+        self.data = somatic._wt5.get_data_readonly()
+        self.axis.set_allowed_values(self.data.axis_expressions)
 
     def on_data_file_written(self):
-        data = somatic._wt5.get_data_readonly()
+        try:
+            for dset in self.data.values():
+                dset.id.refresh()
+        except TypeError:
+            # happens when done writing
+            self.on_data_file_created()
         last_idx_written = somatic._wt5.last_idx_written
         self.idx_string.write(str(last_idx_written))
-        if data is None or last_idx_written is None:
+        if self.data is None or last_idx_written is None:
             return
         # data
         idx = last_idx_written
-        axis = next(a for a in data.axes if a.expression == self.axis.read())
-        channel = data[self.channel.read()]
+        axis = [a for a in self.data.axes if a.expression == self.axis.read()][-1]
+        if f"{axis.expression}_points" in self.data:
```

Comment:
I strongly prefer not using points. Why was this change made?

Reply:
because points are filled out with the points at the outset, which is used to set the window for the plot, without points, it does the weird expanding window thing.

Reply:
I prefer the expanding window thing. I do not want to misrepresent the data by pretending that the hardware has arrived at it's desired destination.

Reply:
If you want to get limits from the points array, I could see that working. The actual xi array should be sourced from the actual hardware positions I think.

Reply:
I can agree to that, I think